hkmc2/shared/src/test/mlscript-compile/FingerTreeList.mls

```diff
@@ -0,0 +1,253 @@
+import "./Option.mls"
+import "./Iter.mls"
+
+open Option
+open Iter
+open FingerTreeList
+
+fun normIdx(idx, len) =
+  if idx < 0 then idx + len
+  else idx
+
+// Follows the semantics of Array.prototype.slice indices:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
+fun normIdxSlice(i, len) =
+  if i < 0 and
+    i >= -len then len + i
+    else 0
+  else i
+
+// Node[T]
+class Branch2(val size, val e1, val e2)
+class Branch3(val size, val e1, val e2, val e3)
```

Comment:
For this and the ones below, why not use code instad of comments? (It's not checked yet, but will be later.) Also, you can use `data class`.
```suggestion
type Node[T] = Branch2[T] | Branch3[T]
data class Branch2(size: Int, e1: ..., e2: ...)
...etc
```